src/main/java/com/williamfiset/algorithms/datastructures/balancedtree/TreapTree.java

```diff
@@ -0,0 +1,175 @@
+/**
+ * This file contains an implementation of a Treap Class. Any comparable data is allowed within this
+ * tree(numbers, strings, comparable Objects, etc...) this is a max heap implementation(highest
```

Comment:
Would it be easy to allow the user to provide any custom Comparator?

Reply:
Hm right now it just extends a generic type T, was this suggestion to implement a custom comparator? Sorry if I don't completely understand

Reply:
The type class takes as input a T extends Comparable<T>, so it's already comparable. I was suggesting allowing a custom comparator for flexibility. The comparator of the class is sufficient for now. We can revisit if there's a use case for it.